spec/models/service_retire_task_spec.rb

```diff
@@ -47,26 +48,49 @@
         miq_request.approve(approver, reason)
       end
 
-      context "resource lacks type" do
-        it "creates service retire subtask" do
-          resource = FactoryBot.create(:service_resource, :resource_type => nil, :service_id => service.id, :resource_id => vm.id)
-          service.service_resources << resource
-          service_retire_task.after_request_task_create
+      it "creates service retire subtask" do
+        service.add_resource!(FactoryBot.create(:service_orchestration))
+        service_retire_task.after_request_task_create
 
-          expect(service_retire_task.description).to eq("Service Retire for: #{service.name} - ")
-          expect(ServiceRetireTask.count).to eq(1)
-        end
+        expect(service_retire_task.description).to eq("Service Retire for: #{service.name} - ")
+        expect(ServiceRetireTask.count).to eq(2)
```

Comment:
Why 2?

Reply:
Because the top level service has one too.

Reply:
Oh, so one for the `Service` and one for the `ServiceOrchestration`?

Reply:
Yup!